wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheMonitor.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin.cache;
+
+/**
+ * Interface for cache health monitors. This class uses a background thread to monitor cache
+ * cluster health and manage state transitions between HEALTHY, SUSPECT, and DEGRADED states.
+ */
+public interface CacheMonitor extends Runnable {
+
+  /**
+   * Registers a cache cluster for monitoring.
+   * A cluster consists of a read-write endpoint and optionally a separate read-only endpoint.
+   *
+   * @param rwEndpoint the read-write endpoint (hostname:port), required
+   * @param roEndpoint the read-only endpoint (hostname:port), can be null if same as RW
+   * @param connection the cache connection for executing health checks
+   */
+  void registerCluster(String rwEndpoint, String roEndpoint, CacheConnection connection);
+
+  /**
+   * Gets the current health state of a cache cluster.
+   * Computes cluster state based on both RW and RO endpoints:
+   * - HEALTHY: both endpoints are healthy
+   * - DEGRADED: either or both endpoints are degraded
+   * - SUSPECT: either or both endpoints are suspect
+   *
+   * @param rwEndpoint the read-write endpoint (hostname:port)
+   * @param roEndpoint the read-only endpoint (hostname:port), can be null
+   * @return the computed cluster health state
+   */
+  HealthState getClusterState(String rwEndpoint, String roEndpoint);
+
+  /**
+   * Reports a cache operation error to trigger state transitions.
+   *
+   * @param endpoint the cache cluster endpoint (hostname:port)
+   * @param error the exception that occurred during the cache operation
+   */
+  void reportError(String endpoint, Exception error, String operation);
```

Comment:
pass in `Throwable`, not `Exception`.

Reply:
Also pass in both read/write endpoints to uniquely identify the cluster, and pass in information for which endpoint is reporting error.